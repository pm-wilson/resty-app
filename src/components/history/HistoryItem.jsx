import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ url, key, method }) => (
  <div key={key} className="history-item">
    <span className="method-area">{method}</span>
    <span className="url-area">{url}</span>
  </div>
);

Input.propTypes = {
  url: PropTypes.string.isRequired,
  key: PropTypes.string.isRequired,
  method: PropTypes.func.isRequired,
};

export default Input;
