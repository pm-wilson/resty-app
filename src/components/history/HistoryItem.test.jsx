import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  afterEach(() => cleanup());
  it('renders HistoryItem', () => {
    const { asFragment } = render(<HistoryItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
