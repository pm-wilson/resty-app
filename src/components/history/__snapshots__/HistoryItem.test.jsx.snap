// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders HistoryItem 1`] = `
<DocumentFragment>
  <div
    class="history-item"
  >
    <span
      class="method-area"
    />
    <span
      class="url-area"
    />
  </div>
</DocumentFragment>
`;
